### ISSUE TITLE
Disposition ownership of Snap and/or Flatpak packaging

### ISSUE THREAD

**Issue by kcgen** [dosbox-staging/dosbox-staging#1107]
The following was discussed under 0.76.1 release preparation tasks.  

**@dreamer wrote on Jan 16, 2021:**

Aside of tasks listed in the project, for 0.76.1, I think we should:

- Take over ownership over Snap package - there are several issues with it - serious ones - e.g. no FluidSynth (despite no technical barriers for having it), bad defaults (controllers are blocked by default), home variable override breaking user expectations…
- Fix issues with Flatpak package - they are basically the same issues as with Snap and some additional ones.
- I really would like to have #712 finished and merged to 0.76.1 and master branches before we do next release (Snap package supports bundling bash completion files, Flatpak does not). This PR is in half-finished state - it does not follow our coding convention for shell files (not a deal breaker, but reviewing bash completion files is seriously annoying and it's yet another pain-point). The biggest problem in this PR is missing support for commandline options that would actually be beneficial and I think some support for options that we do not have. I have not reviewed this code thoroughly yet.
- Automate at least 1-2 tasks from the long list of manual tasks that I need to do during the release, so we could check if it worked correctly during the release. ATM my release checklist have ~40 tasks - performing them takes 2-3 hours of manual work on average (if there are no issues). For interested: [dosbox-staging-release.txt](https://github.com/dosbox-staging/dosbox-staging/files/5825385/dosbox-staging-release.txt)
- Ownership over Snap and Flatpak means that release checklist will grow with another set of tasks - honestly I have no idea how many.

Personally, I am working step by step on Meson, because finishing that work would allow me to automate the release more (e.g. I could implement properly working startup tests or fix the problems with injecting git version info). I was planning to first finish meson, then switch to 0.76.1 (so during release we would give notice to packagers: "hey, take notice of a new buildsystem for *next* release and report if you have problem with it"). If someone else has free cycles to work on Snap, or Flatpak, or bash completion, or automation - that would be appreciated.

I intend to use 0.76.1 as an opportunity to communicate to packagers that brand new buildsystem landed in our master and replaced autoconf for upcoming 0.77.0 release - this way we give them some time to prepare for next version, and to communicate any problems with Meson early before the next major release.

If someone wants to push automation tasks, then good candidate would be implementation of automatic uploads of build artefacts on release (this work needs to be implemented and tested in a private fork, we can't generate numerous test releases in the official repo - I was experiencing). Please ask about details before running with it though…

**Comment by kcgen** [comment]
My only concern is adding more nuanced manual checks to the release process, which will increasing the odds of human error. 

If we want this, then suggest only accepting these PR(s) if the work is accompanied by exact release-prep steps and/or commands,  such as in: https://github.com/dosbox-staging/dosbox-staging/wiki/Maintenance-and-Release-Notes.

**Comment by kcgen** [comment]
Maintaining repo packages such as snap and flatpak packages is currently out of scope.  If we have maintainers willing to step forward and can commit to supporting these packages for user on an ongoing basis-- then great! Closing this for now.
